### PR TITLE
feat(service-portal): Housing benefits frontend + api

### PR DESCRIPTION
**Pull request by thordurhhh**
## What

Add housing benefits api and frontend

https://service-portalhb-frontend-beta.dev01.devland.is/minarsidur/fjarmal/greidslur/husnaedisbaetur

## Why

New feature ✨ 

## Screenshots / Gifs

Screenshot 2024-03-25 at 10 56 06

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Formatting passes locally with my changes
- [ ] I have rebased against main before asking for a review

## Summary by CodeRabbit

- **New Features**
	- Introduced a new query interface for housing benefits payment history with detailed filtering options including date ranges, page numbers, and payment origins.
	- Added a GraphQL resolver for managing queries related to housing benefits, enhancing data retrieval and security with authentication guards.
	- Implemented new GraphQL object types for housing benefit payments, supporting detailed payment information and pagination.
	- Enhanced the service portal with components for displaying housing benefits payments, including tables with pagination, total payment calculations, and data filtering options.
	- Integrated Icelandic locale support for date formatting in the service portal.
	- Provided functionality for exporting housing benefits payment data to CSV or Excel files, along with print capabilities.
	- Added comprehensive localization support for housing benefits-related text in the finance module of the service portal.

- **Style**
	- Implemented styling for new components related to housing benefits payments, ensuring a consistent and user-friendly interface.

- **Documentation**
	- Included detailed documentation for new functions and components, improving code maintainability and developer onboarding.

### CODE REVIEW

File: libs/service-portal/finance/src/screens/HousingBenefits/HousingBenefits.tsx [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+          )}
+          {data?.housingBenefitPayments?.data &&
+            data.housingBenefitPayments.data.length > 0 && (
+              <HousingBenefitsTable
+                setPage={setPage}
+                page={page}
+                payments={data.housingBenefitPayments}
+              />
+            )}
+        </Box>
```
Comment (thorkellmani) [inline libs/service-portal/finance/src/screens/HousingBenefits/HousingBenefits.tsx:133]: There is an `<EmptyTable /> `component if you want it

**Review by thorkellmani — APPROVED** [review]
LGTM; one minor comment

**Comment by busla** [comment]
testing an AI bot, feel free to dismiss any suggestions made.

File: libs/api/domains/housing-benefits/src/lib/housing-benefits.resolver.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  @Query(() => HousingBenefitPaymentsResponse, {
+    name: 'housingBenefitPayments',
+    nullable: true,
+  })
+  @Audit()
+  async getHousingBenefitPayments(
+    @CurrentUser() user: User,
+    @Args('input') input: HousingBenefitPaymentsInput,
+  ): Promise<HousingBenefitPaymentsResponse | null> {
+    return await this.housingBenefitsService.hmsPaymentHistory(user, {
```
Comment (saevarma) [inline libs/api/domains/housing-benefits/src/lib/housing-benefits.resolver.ts:35]: ~ Redundant `return await`

File: libs/api/domains/housing-benefits/src/lib/housing-benefits.resolver.ts [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+@Scopes(ApiScope.financeOverview)
+@Resolver()
+@Audit({ namespace: '@island.is/api/hms-housing-benefits' })
+export class HousingBenefitsResolver {
+  constructor(
+    private housingBenefitsService: HmsHousingBenefitsClientService,
+  ) {}
+
+  @Query(() => HousingBenefitPaymentsResponse, {
+    name: 'housingBenefitPayments',
```
Comment (saevarma) [inline libs/api/domains/housing-benefits/src/lib/housing-benefits.resolver.ts:27]: ~ Minor details but missing 's' at the end to match exact api domain name 😅. And I noticed now that it is missing in the type names as well.
```suggestion
    name: 'housingBenefitsPayments',
```

File: libs/api/domains/housing-benefits/src/lib/models/housingBenefit.model.ts [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+export class HousingBenefitPageInfo {
+  @Field({ nullable: true })
+  hasNextPage!: boolean
+
+  @Field({ nullable: true })
+  hasPreviousPage?: boolean
+}
+
+@ObjectType()
+export class HousingBenefitPayment {
```
Comment (saevarma) [inline libs/api/domains/housing-benefits/src/lib/models/housingBenefit.model.ts:14]: All fields optional 🤔

**Review by saevarma — APPROVED** [review]
🚀

File: libs/api/domains/housing-benefits/src/lib/models/housingBenefit.model.ts (hunk 4 above)
Comment (thordurhhh) [reply libs/api/domains/housing-benefits/src/lib/models/housingBenefit.model.ts]: That's how it is generated from the service 😞 
I'll ask them if they can change it 🤝

**Pull request merged**